app/src/main/java/com/getsensibill/spendmanager/demo/subclass/SubclassKotlinActivity.kt

```diff
@@ -0,0 +1,119 @@
+ *
+ * If you are using your own view, then its important to override the [callSetContentView] and
+ * the [webContainerId] so that the [WebUiActivity] knows where to place the [WebUiFragment].
+ */
+class SubclassKotlinActivity : WebUiActivity(), WebUiFragment.Listener {
+    private lateinit var binding: ActivitySubclassKotlinBinding
+
+    /**
+     * Not required override.
+     *
+     * Can be used to override the setContentView, so that it uses your provided
+     * activities view, instead of the default [WebUiActivity] view.
+     */
+    override fun callSetContentView() {
+        binding = ActivitySubclassKotlinBinding.inflate(layoutInflater)
+        setContentView(binding.root)
+    }
+
+    /**
+     * Not required override, unless you are overriding [callSetContentView]
+     *
+     * If we are overriding the setContentView, we also need to set the container id
+     * which will contain the [WebUiFragment].
+     */
+    override val webContainerId: Int
+        get() = binding.webContainer.id
+
+    /**
+     * Not required override.
+     *
+     * If you want to change the default navigation when first loading into Spend Manager,
+     * you can override this and provide a different [NavigationIntent]. Alternatively,
+     * you can also apply this as an argument when overriding [createWebUiFragment].
+     */
+    override val webInitialNavigation: NavigationIntent
+        get() = NavigationIntent.DASHBOARD
+
+    /**
+     * Not required override.
+     *
+     * Used to create the [WebUiFragment] held in the Activity. Common reason for overriding
+     * this would be if there are arguments that are needed to be supplied to the fragment.
+     */
+    override fun createWebUiFragment(): WebUiFragment = WebUiFragment().apply {
+        arguments = Bundle().apply {
+            // Pass in a navigation override. Defaults as .DASHBOARD
+            putParcelable(WebUiFragment.ARG_NAVIGATION_OVERRIDE, NavigationIntent.DASHBOARD)
+            // Pass in a custom Theme override. Defaults to null
+            putParcelable(WebUiFragment.ARG_PROGRAMMATIC_THEME_OVERRIDE, ProgrammaticTheme(Brand()))
+        }
+    }
+
+    /**
+     * Not required override.
+     *
+     * The shouldNavigateBack from the [webFragment] will inform you if your activity
+     * should handle the back pressed normally.
+     */
+    override fun onBackPressed() {
+        if (webFragment?.shouldNavigateBack() != false) finish()
+    }
+
+    /**
+     * Not required override.
+     *
+     * This will be called when ending the WebUi flow, and will give you a [UiFinishReason]
+     * which can be used to identify why the flow is ending. Default behaviour is to try and
+     * reload the Spend Manager. If you do not want it to attempt an immediate reload of
+     * Spend Manager, you can override this call.
+     */
+    override fun onRequestFinish(reason: UiFinishReason) {
```

Comment:
Should we do a `super` call here so that the activity will still finish?

Reply:
fixed